runelite-client/src/main/java/net/runelite/client/plugins/statustrayicon/StatusTrayIconPlugin.java

```diff
@@ -0,0 +1,164 @@
+
+	private static final int HEIGHT = 42;
+	private static final int WIDTH = 42;
+	private static final int HALF_WIDTH = WIDTH / 2;
+	private static final int MAC_HEIGHT_OFFSET = 4;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientUI clientUI;
+
+	@Inject
+	private StatusTrayIconConfig config;
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN || !(config.showHitpoints() || config.showPrayer()))
+		{
+			return;
+		}
+
+		final int maxHealth = client.getRealSkillLevel(Skill.HITPOINTS);
+		final int currentHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
+		final int scaledMissingHealth = (int) Math.ceil((Math.max(maxHealth - currentHealth, 0) * HEIGHT * 1.0) / maxHealth);
+
+		final int maxPrayer = client.getRealSkillLevel(Skill.PRAYER);
+		final int currentPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
+		final int scaledMissingPrayer = (int) Math.ceil((Math.max(maxPrayer - currentPrayer, 0) * HEIGHT * 1.0) / maxPrayer);
+
+		final BufferedImage hpImage = new BufferedImage(WIDTH, HEIGHT + getYOffset(), BufferedImage.TYPE_4BYTE_ABGR);
+		final Graphics2D g = hpImage.createGraphics();
+
+		if (config.showHitpoints())
+		{
+			g.setColor(REMAINING_HEALTH_COLOR);
+			g.fillRect(0, getYOffset(), config.showPrayer() ? HALF_WIDTH : WIDTH, HEIGHT);
+
+			g.setColor(MISSING_HEALTH_COLOR);
+			g.fillRect(0, getYOffset(), config.showPrayer() ? HALF_WIDTH : WIDTH, scaledMissingHealth);
+		}
+
+		if (config.showPrayer())
+		{
+			g.setColor(REMAINING_PRAYER_COLOR);
+
+			int prayerX = config.showHitpoints() ? HALF_WIDTH : 0;
+			int prayerWidth = config.showHitpoints() ? HALF_WIDTH : WIDTH;
+
+			g.fillRect(prayerX, getYOffset(), prayerWidth, HEIGHT);
+
+			g.setColor(MISSING_PRAYER_COLOR);
+			g.fillRect(prayerX, getYOffset(), prayerWidth, scaledMissingPrayer);
+		}
+
+		if (config.numberToDisplay() !=  StatusNumberMode.NONE)
+		{
+			int numberToDisplay = config.numberToDisplay() == StatusNumberMode.HITPOINTS ? currentHealth : currentPrayer;
+			g.setFont(numberToDisplay < 100 ? STATUS_FONT : STATUS_FONT_SMALL);
+			g.setColor(TEXT_COLOR);
+			int numberDrawY = (numberToDisplay < 100 ? 34 : 30) +  getYOffset();
+			g.drawString(String.valueOf(numberToDisplay), 2, numberDrawY);
+		}
+
+		g.dispose();
+
+		clientUI.getTrayIcon().setImage(hpImage);
+	}
+
+	@Subscribe
+	public void onGameStateChanged(final GameStateChanged event)
+	{
+		if (event.getGameState() != GameState.LOGGED_IN)
+		{
+			resetTrayIcon();
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		resetTrayIcon();
+	}
+
+	@Provides
+	StatusTrayIconConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(StatusTrayIconConfig.class);
+	}
+
+	private void resetTrayIcon()
+	{
+		clientUI.getTrayIcon().setImage(ClientUI.ICON);
+
+	}
+
+	private int getYOffset()
+	{
+		return OSType.getOSType() == OSType.MacOS ? MAC_HEIGHT_OFFSET : 0;
```

Comment:
Why?

Reply:
On mac, the way that `SystemTray` sets up the icon pushes it up against the top of the screen instead of centering it as the other icons are. Windows doesn't have this issue so I added a Mac specific offset.

Reply:
This is the type of information which should be in a comment either on the method here or on the field `MAC_HEIGHT_OFFSET`